src/CoreWCF.WebHttp/src/CoreWCF/Channels/WebMessageEncoderFactory.cs

```diff
@@ -0,0 +1,436 @@
+        {
+            _messageEncoder = new WebMessageEncoder(writeEncoding, maxReadPoolSize, maxWritePoolSize, quotas, contentTypeMapper, javascriptCallbackEnabled);
+        }
+
+        public override MessageEncoder Encoder
+        {
+            get { return _messageEncoder; }
+        }
+
+        public override MessageVersion MessageVersion
+        {
+            get { return _messageEncoder.MessageVersion; }
+        }
+
+        internal static string GetContentType(string mediaType, Encoding encoding)
+        {
+            string charset = TextEncoderDefaults.EncodingToCharSet(encoding);
+            if (!string.IsNullOrEmpty(charset))
+            {
+                return string.Format(CultureInfo.InvariantCulture, "{0}; charset={1}", mediaType, charset);
+            }
+            return mediaType;
+        }
+
+        internal class WebMessageEncoder : MessageEncoder
+        {
+            private const string _defaultMediaType = "application/xml";
+            private WebContentTypeMapper _contentTypeMapper;
+            private string _defaultContentType;
+
+            // Double-checked locking pattern requires volatile for read/write synchronization
+            private volatile MessageEncoder _jsonMessageEncoder;
+            private int _maxReadPoolSize;
+            private int _maxWritePoolSize;
+
+            // Double-checked locking pattern requires volatile for read/write synchronization
+            private volatile MessageEncoder _rawMessageEncoder;
+            private XmlDictionaryReaderQuotas _readerQuotas;
+
+            // Double-checked locking pattern requires volatile for read/write synchronization
+            private volatile MessageEncoder _textMessageEncoder;
+            private object _thisLock;
+            private Encoding _writeEncoding;
+            //bool javascriptCallbackEnabled;
+
+            public WebMessageEncoder(Encoding writeEncoding, int maxReadPoolSize, int maxWritePoolSize, XmlDictionaryReaderQuotas quotas, WebContentTypeMapper contentTypeMapper, bool javascriptCallbackEnabled)
+            {
+                if (writeEncoding == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(writeEncoding));
+                }
+
+                _thisLock = new object();
+
+                TextEncoderDefaults.ValidateEncoding(writeEncoding);
+                _writeEncoding = writeEncoding;
+
+                _maxReadPoolSize = maxReadPoolSize;
+                _maxWritePoolSize = maxWritePoolSize;
+                _contentTypeMapper = contentTypeMapper;
+                //this.javascriptCallbackEnabled = javascriptCallbackEnabled;
+
+                _readerQuotas = new XmlDictionaryReaderQuotas();
+                quotas.CopyTo(_readerQuotas);
+
+                _defaultContentType = GetContentType(_defaultMediaType, writeEncoding);
+            }
+
+            public override string ContentType
+            {
```

Comment:
Expression body style properties for several in this class

Reply:
This is addressed in f4f9310a6b3ef5b0475a2f8d7f184e9f57afce85.